init/eessi_archdetect.sh

```diff
@@ -114,22 +115,28 @@ cpupath(){
         if [ "${cpu_vendor}x" == "${arch_spec[1]}x" ]; then
             # each flag in this CPU specification must be found in the list of flags of the host
             check_allinfirst "${cpu_flags[*]}" ${arch_spec[2]} && best_arch_match=${arch_spec[0]} && \
-                log "DEBUG" "cpupath: host CPU best match updated to $best_arch_match"
+                all_arch_matches="$best_arch_match:$all_arch_matches" && \
+                log "DEBUG" "cpupath: host CPU best match updated to $best_arch_match" 
         fi
     done
   
     log "INFO" "cpupath: best match for host CPU: $best_arch_match"
-    echo "$best_arch_match"
+    if [ "allx" == "${1}x" ]; then
+        echo "$all_arch_matches"
+    else
+        echo "$best_arch_match"
+    fi
 }
 
 # Parse command line arguments
-USAGE="Usage: eessi_archdetect.sh [-h][-d] <action>"
+USAGE="Usage: eessi_archdetect.sh [-h][-d][-a] <action>"
 
-while getopts 'hdv' OPTION; do
+while getopts 'hdva' OPTION; do
     case "$OPTION" in
         h) echo "$USAGE"; exit 0;;
         d) LOG_LEVEL="DEBUG";;
         v) echo "eessi_archdetect.sh v$VERSION"; exit 0;;
+        a) all="all";;
```

Comment:
Ideally we also have a constant like `$CPUARCH_RESULT_ALL` (where `"cpuarch_result_all"` is the value) than we can use for both setting and checking, to avoid bugs due to silly typos (I guess we should also try adding `set -e` so use of unset environment variables are treated like errors)

Reply:
Not quite sure what you mean here, but I think it is addressed in https://github.com/EESSI/software-layer/pull/264/commits/7db0616a0c78f40864507d77085817af6eb7028a